packages/kbn-alerts-ui-shared/src/common/hooks/use_load_rule_type_aad_template_fields.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { EcsFlat } from '@elastic/ecs';
+import { ActionVariable } from '@kbn/alerting-types';
+import type { HttpStart } from '@kbn/core-http-browser';
+import { useQuery } from '@tanstack/react-query';
+import { fetchRuleTypeAadTemplateFields, getDescription } from '../apis';
+
+export interface UseLoadRuleTypeAadTemplateFieldProps {
+  http: HttpStart;
+  ruleTypeId: string;
+  enabled: boolean;
+}
+
+export const useLoadRuleTypeAadTemplateField = (props: UseLoadRuleTypeAadTemplateFieldProps) => {
+  const { http, ruleTypeId, enabled } = props;
+
+  const queryFn = () => {
+    return fetchRuleTypeAadTemplateFields({ http, ruleTypeId });
+  };
+
+  const {
+    data = [],
+    isLoading,
+    isFetching,
+    isInitialLoading,
+  } = useQuery({
+    queryKey: ['useLoadRuleTypeAadTemplateField', ruleTypeId],
+    queryFn,
+    refetchOnWindowFocus: false,
+    enabled,
+  });
+
+  return {
+    data: data.map<ActionVariable>((d) => ({
```

Comment:
In this article [here](https://tkdodo.eu/blog/react-query-data-transformations#3-using-the-select-option) (RQ contributor) it is suggested to use the `select` property to do data transformations for optimization and partial subscriptions. What do you thing of using the same here?